src/Symfony/Component/ErrorHandler/ErrorRenderer/ErrorRenderer.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\ErrorHandler\ErrorRenderer;
+
+use Psr\Container\ContainerInterface;
+use Symfony\Component\ErrorHandler\Exception\FlattenException;
+
+/**
+ * Renders an Exception that represents a Response content.
+ *
+ * @see ErrorRendererInterface
+ */
+class ErrorRenderer
+{
+    private $container;
+
+    public function __construct(ContainerInterface $container)
+    {
+        $this->container = $container;
+    }
+
+    /**
+     * Renders an Exception and returns the Response content.
+     *
+     * @param \Exception|FlattenException $exception An \Exception or FlattenException instance
+     * @param string                      $format    The request format (html, json, xml, etc.)
+     *
+     * @return string The Response content as a string
+     *
+     * @throws \InvalidArgumentException when the renderer does not exist
+     */
+    public function render($exception, string $format = 'html'): string
+    {
+        if (!$this->container->has($format)) {
```

Comment:
this should not be `'error_handler.renderer.'.$format` ?

Reply:
It'll be the `ErrorRendererInterface::getFormat()`, ref: https://github.com/symfony/symfony/pull/31065/files#diff-ca872077b071c60c06b91a837070daabR47

Reply:
BTW, similar approach is done for `FragmentRenderer` feature:
https://github.com/symfony/symfony/blob/238f844c72589020767a0c5928e7b6024c598014/src/Symfony/Component/HttpKernel/DependencyInjection/FragmentRendererPass.php#L57